National Diet Library Online.js

```diff
@@ -0,0 +1,811 @@
+	for (var i=0,ilen=zenkakuNums.length; i<ilen; i++) {
+		zenkakuNums[i] = new RegExp(zenkakuNums[i], "g");
+	}
+	return zenkakuNums;
+}();
+
+function getItemType(details) {
+	var ndlType = details.risMaterialType[0];
+	var itemType = typeMap[ndlType] ? typeMap[ndlType] : "report";
+	return itemType;
+}
+
+function convertImperialDate(dateStr) {
+	var eraOffsetMap = {
+		"\u660E\u6CBB": 1867,
+		"\u5927\u6B63": 1911,
+		"\u662D\u548C": 1925,
+		"\u5e73\u6210": 1988
+	};
+	// 元年
+	dateStr = dateStr.replace("\u5143\u5E74", "1\u5E74");
+	for (var i=0,ilen=zenkakuRex.length; i<ilen; i++) {
+		dateStr = dateStr.replace(zenkakuRex[i], i);
+	}
+	// 明治|大正|昭和|平成...年...月...日
+	var m = dateStr.match(/(\u660E\u6CBB|\u5927\u6B63|\u662D\u548c|\u5e73\u6210)([0-9]+)\u5e74(?:([0-9]+)(?:\u6708([0-9]+)\u65e5)*)*/);
+	if (m) {
+		var era = m[1];
+		var year = (parseInt(m[2], 10) + eraOffsetMap[m[1]]);
+		if (!isNaN(year)) {
+			var dateStr = [year, m[3], m[4]]
+				.filter(function(elem){
+					return elem;
+				}).map(function(elem){
+					while (elem.length < 2) {
+						elem = "0" + elem;
+					}
+					return elem;
+				}).join("-");
+		}
+	}
+	return dateStr;
+}
+
+var fields = [
+	{ ndl: "language", conv: "language" },
+	{ ndl: "title", conv: "title" },
+	{ ndl: "volumeDate", conv: "date" },
+	{ ndl: "pageRange", conv: "pages" },
+	{ ndl: "degreeGrantor", conv: "university" },
+	{ ndl: "degreeName", conv: "thesisType" },
+	{ ndl: "dateGranted", conv: "date" },
+	{ ndl: "dissertationno", conv: "archiveLocation" },
+	{ ndl: "publicationDate", conv: "date" },
+	{ ndl: "scale", conv: "scale" },
+	{ ndl: "edition", conv: "edition" },
+	{ ndl: "seriesTitle", conv: "seriesTitle" },
+	{ ndl: "standardno", conv: "number" },
+	{ ndl: "doi", conv: "DOI" },
+	{ ndl: "isbn", conv: "ISBN" }
+];
+
+function setFields(details, item, altLang) {
+	for (var field of fields) {
+		if (details[field.ndl]) {
+			var val = details[field.ndl][0];
+			// remove enclosing parens
+			val = val.replace(/^\((.*)\)$/, "$1");
+			// split and multilingualize
+			var valSplit = val.split(/\s+=\s+/);
+			val = valSplit[0];
+			if (valSplit[1]) {
+				item.multi._keys[field.conv] = {};
+				item.multi._keys[field.conv][altLang] = valSplit[1];
+			}
+			item[field.conv] = val;
+		}
+	}
+}
+
+function sniffEditor(details) {
+	var creatorType = "author";
+	if (details.responsibility) {
+		for (var r of details.responsibility) {
+			if (r.match(/(editor|edited)/i)) {
+				creatorType = "editor";
+			}
+		}
+	}
+	return creatorType;
+}
+
+function scrape(jsonTxt) {
+	var obj = JSON.parse(jsonTxt);
+	// Uncomment to dump the data object in all its glory
+	// Zotero.debug(JSON.stringify(obj, null, 1));
+	obj = obj.response.docs[0];
+	var details = obj.view_detailParameters;
+
+	var itemType = getItemType(details);
+	var item = new Zotero.Item(itemType);
+
+	var altLang;
+	if (item.language === "jpn") {
```

Comment:
The `item` is just defined two lines before as a new Zotero item. How can then the `item.language` field be set?